### PR TITLE
[bug 1121641] Implement password reset

### PR DESCRIPTION
**Pull request by rehandalal**
The last commit is the only one we care about. The others are from the login PR.

r?

### CODE REVIEW

File: app/js/authentication_controller.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        errors.username);
+
+      update_error_list(document.getElementById('password_errors'),
+        errors.password);
+
+      Utils.toggle_spinner();
+    });
+  }
+
+  function request_password_reset(evt) {
```
Comment (Rik) [inline app/js/authentication_controller.js]: I'd prefer to keep controllers simple. We're being bitten by questions_controller.js having too much logic. So a password_reset_controller.js makes sense to me. And you can move update_error_list to utils.js ? (although I hate files that are just a grab bag of functions)

File: app/js/user.js [hunk 2]
```diff
@@ -120,13 +133,15 @@
...
-        });
-      });
+      return SumoDB.update_user(user_data).then(sync_user);
+    },
+
+    authenticate_user: function(username, password) {
+      return SumoDB.get_token(username, password);
+    },
+
+    request_password_reset: function(username) {
```
Comment (Rik) [inline app/js/user.js]: No need to create this proxy function. We're not dealing with user data so we can call SumoDB directly.

File: app/password_reset.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+<head>
+  <meta charset="utf-8" />
+  <meta name="description" content="SUMO real time support app for FirefoxOS" />
+  <meta name="viewport" content="width=device-width" />
+
+  <title>Welcome to BuddyUP</title>
+
+  <link rel="stylesheet" type="text/css" href="css/building_blocks/layout.css" media="screen" />
+  <link rel="stylesheet" type="text/css" href="css/building_blocks/headers.css" media="screen" />
+  <link rel="stylesheet" type="text/css" href="css/building_blocks/lists.css" media="screen" />
```
Comment (Rik) [inline app/password_reset.html]: I don't think we need this CSS but I haven't checked.

File: app/js/authentication_controller.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+      Utils.refresh_error_list(
+        document.getElementById('password_errors'),
+        errors.password);
+
+      Utils.toggle_spinner();
+    });
+  }
+
+  function register_forms() {
```
Comment (Rik) [inline app/js/authentication_controller.js]: singular now :)

Comment (rehandalal) [inline app/js/authentication_controller.js]: i got that in the register PR

**Pull request merged**